bcgov/HOUS-permit-portal
# Feature/bphh 1988/revision pdf file generation

## Description
- Regenerate the zip file when a permit application is resubmitted
- Generate permit applications and step code pdf files for each submission
- Adds a fix to the submission download modal to include only files that have been completed, i.e. submitted not just uploaded


## What type of PR is this? (check all applicable)

- [ x] 🍕 Feature

## Related Tickets & Documents
https://hous-bssb.atlassian.net/browse/BPHH-1988


## Steps to QA
- Prerequisite: ensure both db migration and data migrations are run
- As a submitter, start a permit application. Ensure there is an energy step code requirement to test the generation of the step code checklist
- Complete and submit the permit application
- As a reviewer request revisions and send to submitter
- As a submitter, make changes to the above permit application and resubmit
- The download modal and zip file should have all the files and generated PDFs for both submissions
- The download modal should also only show files which have been submitted

![Screenshot 2024-08-23 at 10 05 46 AM](https://github.com/user-attachments/assets/324dbfbd-640a-427b-a200-61886ae5f479)
